app/screens/apps_form/apps_form_component.tsx

```diff
@@ -153,24 +144,58 @@ function AppsFormComponent({
             undefined,
             intl.formatMessage({id: 'interactive_dialog.submit', defaultMessage: 'Submit'}),
         );
-        base.enabled = submitting;
+        base.enabled = !submitting;
         base.showAsAction = 'always';
         base.color = theme.sidebarHeaderTextColor;
         return base;
-    }, [theme.sidebarHeaderTextColor, intl, Boolean(submitButtons)]);
+    }, [theme, Boolean(submitButtons), submitting, intl]);
```

Comment:
Here it is interesting to have the `sidebarHeaderTextColor` on the theme. That property is a base type (a string), so when the theme changes, even if it is a new string, a comparison can be made, and not re-run this if it is the same string.